tests/api-mongo.test.js

```diff
@@ -412,3 +416,26 @@ describe('MongoDB Words', function () {
     });
   });
 });
+
+describe('API Requests For Home Directory "/"', () => {
+  describe('/undefinedRoute', () => {
+    it('should return response status of 404', (done) => {
+      const route = '/undefinedRoute';
+      getApiUrlRoute(route).end((_, res) => {
+        expect(res.status).to.equal(404);
+      });
+      done();
```

Comment:
`done` should be placed inside the `end` callback function.

Right now, you're technically calling `done` without waiting for `res` to return back to the client.